File: cpp/94.二叉树的中序遍历.cs
/*
 * @lc app=leetcode.cn id=94 lang=csharp
 *
 * [94] 二叉树的中序遍历
 */

// @lc code=start
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    IList<int> ilist = new  List<int>();
    public void order(TreeNode root)
    {
        if (root == null) {
            return;
        }
        order(root.left);
        ilist.Add(root.val);
        order(root.right);
    }
    public IList<int> InorderTraversal(TreeNode root) {
        order(root);
        return ilist;
    }
}


// @lc code=end

File: cpp/94.二叉树的中序遍历.cpp
/*
 * @lc app=leetcode.cn id=94 lang=cpp
 *
 * [94] 二叉树的中序遍历
 */

// @lc code=start
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void order(TreeNode* root,vector<int> &res)
    {
        if(!root)
            return;
        order(root->left,res);
        res.push_back(root->val);
        order(root->right,res);
    }

    vector<int> inorderTraversal(TreeNode* root) {
        vector<int> rtn;
        order(root,rtn);
        return rtn;
    }
};
// @lc code=end

